py-rse/configuration.Rmd

```diff
@@ -157,21 +167,25 @@ $ python plotcounts.py jane_eyre.csv --outfile jane_eyre.png
 knitr::include_graphics("figures/py-rse/configuration/jane_eyre_big_labels.png")
 ```
 
-While we've successfully edited the plot,
-there are a couple of issues with this solution.
-The first is that because the `matplotlibrc` file sets the system-wide defaults,
-we've locked ourselves into default big labels for any/all plotting we do in future,
-which may not be desirable.
-Secondly, we ultimately want to package up our Zipf's Law code and make it available
-to other people to use.
-We don't have access to their `matplotlibrc` file,
-it's not a solution that can be scaled up to multiple users.
+This does what we want,
+but is usually the wrong approach.
+Since `matplotlibrc` file sets system-wide defaults,
+we will now have big labels by default for all plotting we do in future,
+which we may not want.
+Secondly,
+we want to package our Zipf's Law code and make it available to other people.
+That package won't include our `matplotlibrc` file,
+and we don't have access to the one on their computer,
+so this solution isn't as reproducible as others.
 
-### Create a matplotlib style sheet
+A global options file *is* useful, though.
+If we are using Matplotlib with [LaTeX][latex] to generate reports
+and the latter is installed in an unusual place on our computing cluster,
+a one-line change in `matplotlibrc` can prevent a lot of failed jobs.
 
-Instead of directly editing the `matplotlibrc` file to change
-the default plot characteristics,
-there are a number of pre-defined styles provided by Matplotlib:
+## Creating a Style Sheet {#configuration-stylesheet}
```

Comment:
Would this heading be better phrased as something like "The User Configuration File" (i.e. the general concept we are teaching) rather than "Creating a Style Sheet" (i.e. something specific to matplotlib)?

My thinking is that in this chapter we basically cover four program configuration options:
1. system-wide/global configuration files
2. user-specific configuration files
3. job-specific configuration files
4. command line options

The other headings (The Global Configuration File, Adding Command-Line Options and
A Job Control File) refer to one of those general options/concepts rather than the specific feature of matplotlib that we're using.

Reply:
Good catch - title changed.